scripts/router/router.sh

```
@@ -0,0 +1,84 @@
+
+sysctl -w net.ipv4.ip_forward=1
+sysctl -w net.ipv6.conf.all.disable_ipv6=0
+sysctl -w net.ipv6.conf.default.disable_ipv6=0
+sysctl -w net.ipv6.conf.all.forwarding=1
+sysctl -w net.ipv6.conf.default.forwarding=1
+
+TEMPLATE_FILE="router.nft"
+CONFIG_FILE="/tmp/router.nft"
+
+# Copy template file to working config
+cp "$TEMPLATE_FILE" "$CONFIG_FILE"
+
+echo "add rule inet router postrouting ip saddr $INTERNAL_NET_V4 oifname \"internet\" masquerade ${MASQUERADE_TYPE:-}" >>"$CONFIG_FILE"
+echo "add rule inet router postrouting ip6 saddr $INTERNAL_NET_V6 oifname \"internet\" masquerade ${MASQUERADE_TYPE:-}" >>"$CONFIG_FILE"
+
+# Add port forwarding rules if specified
+if [ -n "${PORT_FORWARDS:-}" ]; then
+    echo "$PORT_FORWARDS" | tr ',' '\n' | while IFS=' ' read -r port internal_ip protocol; do
+        if [ -z "$port" ] || [ -z "$internal_ip" ] || [ -z "$protocol" ]; then
+            continue
+        fi
+
+        # Determine if internal IP is IPv4 or IPv6 and append rules to config file
+        case "$internal_ip" in
+        *:*) # IPv6 address
+            echo "add rule inet router prerouting ip6 daddr $PUBLIC_IPV6 $protocol dport $port dnat to [$internal_ip]:$port" >>"$CONFIG_FILE"
+            echo "add rule inet router input ip6 daddr $internal_ip $protocol dport $port accept" >>"$CONFIG_FILE"
+            ;;
+        *) # IPv4 address
+            echo "add rule inet router prerouting ip daddr $PUBLIC_IPV4 $protocol dport $port dnat to $internal_ip:$port" >>"$CONFIG_FILE"
+            echo "add rule inet router input ip daddr $internal_ip $protocol dport $port accept" >>"$CONFIG_FILE"
+            ;;
+        esac
+    done
+fi
+
+echo "-----------------------------------------------------------------------------------------------"
+cat "$CONFIG_FILE"
+echo "-----------------------------------------------------------------------------------------------"
+
+nft -f "$CONFIG_FILE"
+
+rm "$CONFIG_FILE"
+
+echo "1" >/tmp/setup_done # Health check marker
```

We may be able to get away with `condition: "service_started"` and no healthcheck here if this script is expected to complete fairly quickly.

---

It was helpful to have the healthcheck whilst working on it to more easily see what failed instead of looking at logs from 5 routers.

---

It doesn't really cost anything so I figured I'd keep it.